template/example_test.go

```diff
@@ -50,44 +50,51 @@ func ExampleTemplate_complexMessage() {
 .input { $color :color style=RGB}
 {{John is { $age } years old and his favorite color is { $color }.}}`
 
-	color := func(value any, options template.Options, locale language.Tag) (any, error) {
+	color := func(value any, options template.Options, locale language.Tag) (*template.ResolvedValue, error) {
+		errorf := func(format string, args ...any) (*template.ResolvedValue, error) {
+			return template.NewResolvedValue(""), fmt.Errorf("exec color function: "+format, args...)
+		}
+
 		if value == nil {
-			return "", fmt.Errorf("input is required: %w", mf2.ErrBadOperand)
+			return errorf("input is required: %w", mf2.ErrBadOperand)
 		}
 
 		color, ok := value.(string)
 		if !ok {
-			return nil, fmt.Errorf("input is not a string: %w", mf2.ErrBadOperand)
+			return errorf("input is not a string: %w", mf2.ErrBadOperand)
 		}
 
 		if options == nil {
-			return color, nil
+			return template.NewResolvedValue(color), nil
```

Comment:
no format?

Reply:
It uses defaultFormat while it still can (until we start throwing errors).

Reply:
We know we'll add erroring later and ResolvedValue MUST have at least one - formatting or selection. Let's make sure we return a valid (error free) ResolvedValue.

Reply:
Added format function